### PR TITLE
Fix svcat version command & add tests

### PR DESCRIPTION
**Pull request by A-Hilaly**
Dealing with #2092 

- [x] Fix version command
- [x] Add tests

Edit: Oops linked wrong issue id

### CODE REVIEW

File: cmd/svcat/versions/version_cmd.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ func (c *versionCmd) version() error {
 		client = pkg.VERSION
 	}
 
+	output.WriteVersion(c.Output, client, "")
```
Comment (A-Hilaly) [inline cmd/svcat/versions/version_cmd.go]: Do you think i should split `output.WriteVersion` into `output.WriteClientVersion` and `output.WriteServervVersion` ?

File: cmd/svcat/output/version.go [hunk 2]
```diff
@@ -24,9 +24,9 @@ import (
 // WriteVersion prints a list of bindings.
 func WriteVersion(w io.Writer, client, server string) {
 	if client != "" {
-		fmt.Fprintf(w, "client: %s\n", client)
+		fmt.Fprintf(w, "Client Version: %s\n", client)
```
Comment (A-Hilaly) [inline cmd/svcat/output/version.go]: Just like kubectl version command:
https://github.com/kubernetes/kubernetes/blob/60cd056cb443c23b53240d274df4d5abc85c7aaf/pkg/kubectl/cmd/version.go#L121-L124

Comment (carolynvs) [reply cmd/svcat/output/version.go]: good idea! Yes that would work well.

File: cmd/svcat/versions/version_cmd.go (hunk 1 above)
Comment (carolynvs) [reply cmd/svcat/versions/version_cmd.go]: good idea! Yes that would work well.

**Comment by carolynvs** [comment]
Due to a bug in our travis file, this PR must be rebased in order to get the CI build to run for this PR.

/label needs-rebase

File: cmd/svcat/versions/version_cmd_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		client     bool
+		server     bool
+		wantOutput string
+		wantError  bool
+	}{
+		{
+			name:       "show client version only",
+			client:     true,
+			server:     false,
+			wantOutput: "Client Version: UNKNOWN\n",
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply cmd/svcat/versions/version_cmd_test.go]: I believe that you can set the version for your tests with the following code:

```
import "github.com/kubernetes-incubator/service-catalog/pkg"

pkg.VERSION = "v0.0.0"
```

File: cmd/svcat/versions/version_cmd_test.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Set up fake data
+			//TODO: feed fakes array
+			var fakes []runtime.Object
+			k8sClient := k8sfake.NewSimpleClientset(fakes...)
+			svcatClient := svcatfake.NewSimpleClientset()
+			output := &bytes.Buffer{}
+			fakeApp, _ := svcat.NewApp(k8sClient, svcatClient, "default")
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply cmd/svcat/versions/version_cmd_test.go]: Let's try something like this (untested):

```
svcatClient.PrependReactor("get", "version",
  func(action testing2.Action) (handled bool, ret runtime.Object, err error) {
    return true, nil, errors.New("sabotaged")
  })

```

I believe that will cause a 500 error to be returned when you ask for the server version. Let me know if that doesn't work!

[the remark this answers is not in the document]
Comment (A-Hilaly) [reply cmd/svcat/versions/version_cmd_test.go]: I tested your suggestion, `fakeApp` still doesn't change behaviour 🤔

**Comment by A-Hilaly** [comment]
@carolynvs the rebase went so bad i guess

**Comment by carolynvs** [comment]
Let me know if you'd like help with the rebase!

File: cmd/svcat/versions/version_cmd_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (carolynvs) [reply cmd/svcat/versions/version_cmd_test.go]: Doh! Let me actually test the code and send you something that works.

**Comment by A-Hilaly** [comment]
I dont know how but it worked 🥇

File: cmd/svcat/versions/version_cmd_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply cmd/svcat/versions/version_cmd_test.go]: Would be great ! Thank you.

**Comment by carolynvs** [comment]
@A-Hilaly Your merge worked just fine 👍 , when your PR is accepted and merged, all the commits will be squashed and just the changes that you see in the Files tab of the PR will end up being included.

Now that the CI build has run on your PR, there is a build error that needs to be addressed:
https://travis-ci.org/kubernetes-incubator/service-catalog/jobs/392480016#L736

**Comment by A-Hilaly** [comment]
@carolynvs Thanks for clarifying ! 
I fixed the changed the header boilerplate.

**Comment by carolynvs** [comment]
@A-Hilaly After digging into the fakes provided by our generated clientset, I've come to realize that it's not possible to fake the server returning an error when querying the `ServerVersion()` at this time.

https://github.com/kubernetes/kubernetes/blob/a9e244d81f05d1fa6965f33b9a3ec04cce6d058f/staging/src/k8s.io/client-go/discovery/fake/discovery.go#L111-L123

Sorry for not investigating if that was possible sooner! 😞 

So the tests you have right now are the best we can do. Let's just remove the commented out test and any TODO's, and ship this. :shipit:

**Comment by A-Hilaly** [comment]
Tests passed now ! i got this wierd error before : 
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

**Pull request merged**